misc-utils/lsfd-sock-xinfo.c

```diff
@@ -52,41 +55,132 @@ static void load_xinfo_from_proc_udp6(ino_t netns_inode);
 static void load_xinfo_from_proc_udplite6(ino_t netns_inode);
 static void load_xinfo_from_proc_raw6(ino_t netns_inode);
 static void load_xinfo_from_proc_netlink(ino_t netns_inode);
+static void load_xinfo_from_proc_packet(ino_t netns_inode);
 
 static int self_netns_fd = -1;
 static struct stat self_netns_sb;
 
 static void *xinfo_tree;	/* for tsearch/tfind */
 static void *netns_tree;
 
+struct iface {
+	unsigned int index;
+	const char *name;
+};
+
+static const char *get_iface_name(ino_t netns, unsigned int iface_index);
+
+static int iface_compare(const void *a, const void *b)
+{
+       const struct iface *iface_a = a;
+       const struct iface *iface_b = b;
+
+       if (iface_a->index < iface_b->index)
+	       return -1;
+       else if (iface_a->index > iface_b->index)
+	       return 1;
+       else
+	       return 0;
+}
+
+static void iface_free(void * iface)
+{
+       free((void *)(((struct iface *)iface)->name));
```

Comment:
Instead of this casting, why not the same pattern as in `compare`? Looks clearer to me.

Reply:
Now the `name` member is not a pointer. It is an in-place array now.